front/pages/api/v1/w/[wId]/assistant/agent_configurations.ts

```diff
@@ -1,24 +1,30 @@
 import type { GetAgentConfigurationsResponseType } from "@dust-tt/client";
-import type { WithAPIErrorResponse } from "@dust-tt/types";
+import type {
+  LightAgentConfigurationType,
+  WithAPIErrorResponse,
+} from "@dust-tt/types";
 import { isLeft } from "fp-ts/lib/Either";
 import * as t from "io-ts";
 import * as reporter from "io-ts-reporters";
 import type { NextApiRequest, NextApiResponse } from "next";
 
 import { getAgentConfigurations } from "@app/lib/api/assistant/configuration";
+import { getAgentsRecentAuthors } from "@app/lib/api/assistant/recent_authors";
 import { withPublicAPIAuthentication } from "@app/lib/api/auth_wrappers";
 import type { Authenticator } from "@app/lib/auth";
 import { apiError } from "@app/logger/withlogging";
 
-const GetAgentConfigurationsViewSchema = t.partial({
+export const GetAgentConfigurationsQuerySchema = t.type({
   view: t.union([
     t.literal("all"),
     t.literal("list"),
     t.literal("workspace"),
     t.literal("published"),
     t.literal("global"),
     t.literal("favorites"),
+    t.undefined,
   ]),
+  withAuthors: t.union([t.literal("true"), t.literal("false"), t.undefined]),
```

Comment:
any reason why not using `t.boolean` ?

Reply:
Just to match with private API

<img width="1030" alt="Screenshot 2024-12-02 at 18 02 18" src="https://github.com/user-attachments/assets/3c0fd5e9-398c-4f0a-b2a8-64044519de0e">

https://github.com/dust-tt/dust/blob/main/types/src/front/api_handlers/internal/agent_configuration.ts#L27